builders/xcm/fees.md

```diff
@@ -0,0 +1,230 @@
+---
+title: XCM Execution Fees
+description: Learn about the XCM instructions involved in handling XCM execution fee payments and how to calculate fees on Polkadot, Kusama, and Moonbeam-based networks.
+---
+
+# XCM Fees on Moonbeam
+
+![XCM Fees Banner](/images/builders/xcm/fees/fees-banner.png)
+
+## Introduction {: #introduction}
+
+XCM aims to be a language that communicates ideas between consensus systems. It is designed to be general, extensible, and efficient so that it remains useful and future-proof throughout a growing ecosystem. As such, the generality applies to concepts including payments of fees for XCM execution. 
+
+In Ethereum, fees are baked into the transaction protocol, whereas in the Polkadot ecosystem each parachain has the flexibility to define how XCM fees are handled. If fees are required, XCM provides the ability to buy execution resources with assets.
+
+This guide will cover aspects of fee payment such as who is responsible to pay XCM execution fees, how it is paid for, and how the fees are calculated on Moonbeam.
+
+## Payment of Fees {: #payment-of-fees }
+
+Consider the following scenario: Alice has some DOT on Polkadot and she wants to transfer it to xcDOT on Moonbeam, to her account name Alith. She sends an XCM message along with a given amount of DOT on Polkadot with instructions to deposit the xcDOT into her Alith account. Part of the instructions are executed on Polkdot and the other part are executed on Moonbeam. 
+
+How does Alice pay Moonbeam to execute these instructions and fulfill her request? It is done through a series of XCM instructions which essentially enable her to buy execution time using the balance of the transferred assets. This means that when Alice sends a set amount of DOT and then when she receives the xcDOT in her Alith account, she receives the transferred amount minus any XCM execution fees.
+
+In general, the process can be described as follows:
+
+1. Some assets need to be provided
+2. The exchange of assets for compute time (or weight) must be negotiated
+3. The XCM operations will be performed as instructed and any leftover assets will be sent to a destination account
+
+### XCM Instructions {: #xcm-instructions }
```

Comment:
I think I would make this section more to the point. Like "As mentioned before, a combination of XCM instructions result in different actions. For example, to move DOT to Moonbeam, the following XCM instructions are needed:

List here

On the other hand, to move `xcDOT` from Moonbeam to Polkadot, the following XCM instructions are neeeded:

List Here

I would also add where is each instruction executed (origin chain or destination chain)